presto-docs/src/main/sphinx/security/password-file.rst

```diff
@@ -36,6 +36,45 @@ Property                             Description
                                      Defaults to ``1000``.
 ==================================== ==============================================
 
+Password Validation
+-------------------
+
+Password validation in Presto supports both `PBKDF2WithHmacSHA256` and `PBKDF2WithHmacSHA1` algorithms. 
+To ensure modern cryptographic standards, clients are encouraged to use `PBKDF2WithHmacSHA256`. 
+A fallback mechanism is available to maintain compatibility with legacy systems using `PBKDF2WithHmacSHA1`.
+
+API Method:
+^^^^^^^^^^^
+The following method uses the `PBKDF2WithHmacSHA256` validation mechanism and includes a fallback mechanism:
+
+.. code-block:: java
+
+    /**
+     * @Deprecated using PBKDF2WithHmacSHA1 is deprecated and clients should switch to PBKDF2WithHmacSHA256
+     */
+    public static boolean doesPBKDF2PasswordMatch(String inputPassword, String hashedPassword)
+    {
+        PBKDF2Password password = PBKDF2Password.fromString(hashedPassword);
+
+        // Validate using PBKDF2WithHmacSHA256
+        if (validatePBKDF2Password(inputPassword, password, "PBKDF2WithHmacSHA256")) {
+            return true;
+        }
+
+        // Fallback to PBKDF2WithHmacSHA1
+        LOG.warn("Using deprecated PBKDF2WithHmacSHA1 for password validation.");
+        return validatePBKDF2Password(inputPassword, password, "PBKDF2WithHmacSHA1");
+    }
+
+**Fallback Mechanism:**
+
+If `PBKDF2WithHmacSHA256` fails for legacy reasons, the system gracefully falls back to `PBKDF2WithHmacSHA1` while logging a warning.
+
+**Important Notes:**
```

Comment:
```suggestion
**Notes:**
```
"Important" is redundant - all notes are important or they wouldn't be notes. 

That said, I suggest you consider if moving these two sentences to after the introductory paragraph - before the API Method header in line 46 - would emphasize these two sentences enough that you don't need this Notes section at all. 

To show what I mean, in a local doc build I combined your first note into the last sentence of the first paragraph as they appeared to say the same thing, and moved the second Note to a stand-alone sentence after the introductory paragraph to emphasize it. Here's how my suggestion looks: 

<img width="742" alt="Screenshot 2024-12-04 at 1 12 24 PM" src="https://github.com/user-attachments/assets/0f7da798-3a03-4a5f-a43f-403b5dd46629">

Think about if this suggestion accurately represents your intention, and if it doesn't, ignore it (to keep the Notes section), or do something else that works better.